Example/Backpack/BPKThemeContainerController.m

```diff
@@ -0,0 +1,72 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "BPKThemeContainerController.h"
+#import "Backpack_Native-Swift.h"
+#import <objc/runtime.h>
+
+@interface BPKThemeContainerController ()
+@end
+
+@implementation BPKThemeContainerController
+
+static BPKThemeContainerController *inst;
+
++ (BPKThemeContainerController *) inst {
+    return inst;
+}
+
+bool additionalSubviewAdded = NO;
+
+- (void) viewDidLoad {
+    [super viewDidLoad];
+
+    inst = self;
+}
+
+- (void) swapThemeContainers {
+    // TODO Get theme from settings once implemented!
+    NSString *themeName = nil;
+
+    NSArray *subviewsToPreserve = [[self view] subviews];
+    if(additionalSubviewAdded){
+        subviewsToPreserve = [[subviewsToPreserve objectAtIndex:0] subviews];
+    }
+
+    UIView *themedView = nil;
+    if ([themeName  isEqual: @"London"]){
+        themedView = [[BPKLondonThemeContainer alloc]initWithFrame:self.view.frame];
+    } else if ([themeName  isEqual: @"Hong Kong"]) {
+        themedView = [[BPKHongKongThemeContainer alloc]initWithFrame:self.view.frame];
+    } else if ([themeName  isEqual: @"Doha"]) {
+        themedView = [[BPKDohaThemeContainer alloc]initWithFrame:self.view.frame];
+    }else{
+        themedView = [[UIView alloc]initWithFrame:self.view.frame];
```

Comment:
A `switch` statement would be neater here IMO.

Reply:
I'm not sure it would, because Obj-C 😆
https://stackoverflow.com/a/18709553

Reply:
Goddamit 😂

Reply:
IKR!